### PR TITLE
improvement: Semantic highlight for using directives

### PR DESCRIPTION
**Pull request by jkciesluk**
Using directives allow values not wrapped in double quotes, and we should highlight them the same way

connected to https://github.com/VirtusLab/scala-cli/issues/2178

### CODE REVIEW

File: metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala [hunk 1]
```diff
@@ -309,45 +309,48 @@ object SemanticTokensProvider {
...
-          convertTokensToIntList(
-            tk.text,
-            delta,
-            tpe,
-            mod,
-          )
-      }
-      buffer.addAll(
-        toAdd
+    var text = directive
```
Comment (kasiaMarek) [inline metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala:321]: So it's not really important but I feels a bit hard to follow how and why delta is changed in this manner.

For me it could be a bit more readable to use @tailrec function instead of foreach, so it's easier to see which args change and how they change. Also instead of `val idx = text.indexOf(part)` before `convertTokensToIntList` you could count the number of ` [,\\s]` after `convertTokensToIntList` to adjust the `text` and `delta` in one place.

I'm not sure it will make things much better so it's up to you to decide if you even want to do anything about it.

**Review by kasiaMarek — APPROVED** [review]

**Review by tgodzik — APPROVED** [review]
LGTM!

File: metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala (hunk 1 above)
Comment (jkciesluk) [reply metals/src/main/scala/scala/meta/internal/metals/SemanticTokensProvider.scala]: I've  changed it to tailrec function, I agree its more readable , thanks for the suggestion!
